application/models/GoogleSheets_Model.php

```diff
@@ -108,6 +108,60 @@ public function addNewRecord($email, $fullName, $paymentMethod)
         $result = $this->service->spreadsheets_values->update($this->spreadsheetId, $range, $body, $params);
     }
 
+    /**
+     * Mark an attendees attendance in the google sheets
+     *
+     * @param string $eventName The name of the event
+     * @param string $email The email of the user to record.
+     * @param string @upi The upi of the user if it exists
+     */
+    public function markAsPresent($eventName, $email = null, $upi = null)
+    {
+        // This should probably be in a try/catch block
+        if (is_null($email) and if_null($upi)) {
+            throw new Exception("You need to enter an email or UPI.");
+            return false;
+        }
+
+        if (is_null($eventName)) {
+            throw new Exception("You need to enter a sheet name.");
+            return false;
+        }
+        
+        //  Navigate to the correct spreadsheet
+        $this->setCurrentSheetName($eventName);
+
+        //  If they do exist, find the cell coordinates where you want to place a 'P'
+        $emailCell = $this->getCellCoordinate($email, 'B');
+
+        list(, $row) = NULL;
+
+        if ($emailCell != NULL)
+            list(, $row) = $this->convertCoordinateToArray($emailCell);
+        else {
+            $upiCell = $this->getCellCoordinate($upi, 'B');
+
+            if ($upiCell != NULL)
+                list(, $row) = $this->convertCoordinateToArray($upiCell);
+            else
+                return false;
+        }
```

Comment:
Since we're using PHP 7.0, we can do something cool using what's called a "Null coalescing operator" (https://www.php.net/manual/en/migration70.new-features.php) 😃 

This means we can set a variable like so...

Example usage:
```
$a = null || "Happy"; // $a will be set to happy
$a = "Happy" || null; // $a will be set to happy
$a = "Sad" || "Happy"; // $a will be set to sad
$a = null || null; // $a will be set to null
```

For this example, we can replace:
```
list(, $row) = NULL;

if ($emailCell != NULL)
    list(, $row) = $this->convertCoordinateToArray($emailCell);
else {
    $upiCell = $this->getCellCoordinate($upi, 'B');

    if ($upiCell != NULL)
        list(, $row) = $this->convertCoordinateToArray($upiCell);
    else
        return false;
}
```
to...
```
// If either $emailCell or $upiCell is defined, then get the row number with priority on email. Otherwise, set $row to null
list(, $row) = ($emailCell || $upiCell) ? $this->convertCoordinateToArray($emailCell ?? $upiCell) : null;

// If row is not defined (i.e. no user row was found), return false
if (!$row) {
    return false;
}

...
```

Have a look at that code example and see if it makes sense, otherwise, let me know tonight! 😄